src/escape.d

```diff
@@ -90,6 +90,13 @@ bool checkParamArgumentEscape(Scope* sc, FuncDeclaration fdc, Parameter par, Exp
                 unsafeAssign(v, "variadic variable");
             }
         }
+        else
+        {
+            /* v is not 'scope', and is assigned to a parameter that may escape.
+             * Therefore, v can never be 'scope'.
+             */
+            v.doNotInferScope = true;
```

Comment:
Why can't this use the prevalent scheme of clearing `STCmaybescope`?
Maybe I'm looking at the commits in the wrong order, the PR is a bit messed up.